lib/app/widgets/account_avatar.dart

```diff
@@ -1,49 +1,73 @@
 import 'package:black_hole_flutter/black_hole_flutter.dart';
-import 'package:flutter/material.dart';
+import 'package:flutter/material.dart' hide Banner;
 import 'package:hive_cache/hive_cache.dart';
 
 import '../data.dart';
+import '../services/banner.dart';
 import '../services/storage.dart';
 import '../utils.dart';
 import 'account_dialog.dart';
 import 'cache_utils.dart';
 
+class AccountButton extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return InkWell(
+      customBorder: CircleBorder(),
+      onTap: () {
+        showDialog(context: context, builder: (_) => AccountDialog());
+      },
+      child: Padding(
+        padding: EdgeInsets.all(8),
+        child: ValueListenableBuilder<Set<Banner>>(
+          valueListenable: services.banners,
+          builder: (context, banners, _) {
+            return banners.contains(Banners.demo)
+                ? Center(child: DemoTag())
+                : AccountAvatar();
+          },
+        ),
+      ),
+    );
+  }
+}
+
+class DemoTag extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return Material(
+      color: Colors.orange.withOpacity(0.5),
+      borderRadius: BorderRadius.circular(8),
+      child: Padding(
+        padding: EdgeInsets.all(4),
+        child: Text(
+          context.s.app_demo.toUpperCase(),
+          style: TextStyle(fontSize: 12, fontWeight: FontWeight.bold),
+        ),
+      ),
+    );
+  }
+}
+
 class AccountAvatar extends StatelessWidget {
   const AccountAvatar();
 
   @override
   Widget build(BuildContext context) {
     return EntityBuilder<User>(
       id: services.storage.userId,
-      builder: handleError((context, user, fetch) {
+      builder: (context, snapshot, fetch) {
         final backgroundColor =
-            user?.avatarBackgroundColor ?? context.theme.primaryColor;
-
+            snapshot.data?.avatarBackgroundColor ?? context.theme.primaryColor;
         return CircleAvatar(
           backgroundColor: backgroundColor,
           maxRadius: 16,
           child: Text(
-            user?.avatarInitials ?? '…',
+            snapshot.data?.avatarInitials ?? snapshot.hasError ? 'X' : '…',
```

Comment:
`??` has a higher precedence than `e1 ? e2 : e3`: https://dart.dev/guides/language/specifications/DartLangSpec-v2.2.pdf#page=199
```suggestion
            snapshot.data?.avatarInitials ?? (snapshot.hasError ? 'X' : '…'),
```